planner/core/find_best_task.go

```diff
@@ -1112,12 +1112,6 @@ func (ds *DataSource) convertToTableScan(prop *property.PhysicalProperty, candid
 		tblColHists:       ds.TblColHists,
 		cst:               cost,
 	}
-	if ts.StoreType == kv.TiFlash {
-		if prop.TaskTp != property.CopTiFlashLocalReadTaskType && prop.TaskTp != property.RootTaskType && prop.TaskTp != property.CopTiFlashGlobalReadTaskType {
-			logutil.BgLogger().Error("invalid task:", zap.String("type", prop.TaskTp.String()))
-			return invalidTask, nil
-		}
-	}
```

Comment:
So it is guaranteed in somewhere else that `CopTiFlashLocalReadTaskType` and `CopTiFlashGlobalReadTaskType` will not be generated with non-TiFlash storeType based datasource?

Reply:
Sometimes CopSingleReadTaskType can generate TiFlash Plan, we should not cut this search branch.

Reply:
OK， but better to return `invalidTask` if storeType is not TiFlash and task type is CopTiFlashxxx

Reply:
added in https://github.com/hanfei1991/tidb/blob/ed3065d287773a654e38e3faa4271417253dbd2e/planner/core/find_best_task.go#L380-L379